### PR TITLE
Make Spinner widget open from start-of-list when nothing is selected

### LINKED ISSUE #1935: Spinner widget opens with end-of-list
#### Software and hardware versions 
Collect v1.13.0 and older versions

#### Problem description
Spinner widget opens with end-of-list

#### Steps to reproduce the problem
ezgif com-video-to-gif 1

#### Expected behavior
The first element should be visible.

#### Other information 
Form for testing: 
[selectOneMinimal.xml.txt](https://github.com/opendatakit/collect/files/1758305/selectOneMinimal.xml.txt)

@opendatakit-bot label "contributor friendly"

### PR DESCRIPTION
**Pull request by iammvaibhav**
Closes #1935 

#### What has been done to verify that this works as intended?
Used the form for testing provided In the Issue

#### Why is this the best possible solution? Were any other approaches considered?
AppCompatSpinner by default doesn't provide such functionality, so in order to achieve this, extending AppCompatSpinner was must.

#### Are there any risks to merging this code? If so, what are they?
I don't think so.

#### Do we need any specific form for testing your changes? If so, please attach one.
[selectOneMinimal.xml.txt](https://github.com/opendatakit/collect/files/1769226/selectOneMinimal.xml.txt)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/views/CustomSpinner.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package org.odk.collect.android.views;
+
+import android.content.Context;
+import android.support.v7.widget.AppCompatSpinner;
+import android.util.AttributeSet;
+
+/**
+ * Created by vaibhav on 1/3/18.
+ */
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/views/CustomSpinner.java]: Please replace this comment with a description of the class. The very top should have a copyright notice with your name (you are the copyright holder) -- https://github.com/opendatakit/collect/blob/master/LICENSE.md#appendix-how-to-apply-the-apache-license-to-your-work

File: collect_app/src/main/java/org/odk/collect/android/views/CustomSpinner.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+public class CustomSpinner extends AppCompatSpinner {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/views/CustomSpinner.java]: What would a more descriptive name be? How about `ScrolledToTopSpinner` or something like that?

File: collect_app/src/main/java/org/odk/collect/android/views/CustomSpinner.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    private boolean toggleFlag = true;
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/views/CustomSpinner.java]: This should also have a more descriptive name. I would call it `spinnerClicked` and have it start as `false`. It's confusing that the condition in `getSelectedItemPosition` is negated.

**Review by lognaturel — CHANGES_REQUESTED** [review]
Thanks, @iammvaibhav! I've made a few comments inline.

**Comment by iammvaibhav** [comment]
Thank You @lognaturel for suggestions. I have made the required changes.

**Comment by iammvaibhav** [comment]
I have no idea why build_debug is failing on CircleCI. The logs aren't helpful either.
I ran the same task `./gradlew pmd checkstyle lint findbugs` on my computer and the build was successful.

Can you please point out any errors?

**Comment by mmarciniak90** [comment]
**Tested with success!**

Verified on Android: 4.1, 4.2, 4.4, 5.1, 6.0, 7.0, 8.0.

Verified cases: 
* new form
* saved form
* both device orientations

New, related issue was created - #1960 

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thanks, @iammvaibhav! The failure was due to the continuous integration server running out of memory. I simply re-ran it and it passed.

**Pull request merged**